### PR TITLE
Add priority sampling

### PR DESCRIPTION
**Pull request by labbati**
### Description

This PR adds priority sampling management to the game. In the process, some refactoring to the way we handle distributed tracing in general was necessary.

### Readiness checklist
- [x] [Changelog entry](docs/changelog.md) added, if necessary
- [x] Tests added for this feature/bug

### CODE REVIEW

File: src/DDTrace/Transport/Http.php [hunk 1]
```diff
@@ -62,19 +66,24 @@ private function configure($config)
     {
         $host = getenv(self::AGENT_HOST_ENV) ?: self::DEFAULT_AGENT_HOST;
         $port = getenv(self::TRACE_AGENT_PORT_ENV) ?: self::DEFAULT_TRACE_AGENT_PORT;
-        $path = self::DEFAULT_TRACE_AGENT_PATH;
-        $endpoint = "http://${host}:${port}${path}";
+        $defaultEndpoint = "http://${host}:${port}" . self::DEFAULT_TRACE_AGENT_PATH;
+        $prioritySamlpingEndpoint = "http://${host}:${port}" . self::PRIORITY_SAMPLING_TRACE_AGENT_PATH;
```
Comment (pawelchcki) [inline src/DDTrace/Transport/Http.php]: I know this is probably not very bad for performance. 
But it worries me a bit that we're doing some computations every request, that are not normally used during the request and after.

Overall would be great to defer most of the transport setup to the very end when we need to actually send  (but this is probably out of the scope for this PR).

File: src/DDTrace/SpanContext.php [hunk 2]
```diff
@@ -113,4 +156,22 @@ private static function nextId()
     {
         return bin2hex(random_bytes(8));
     }
+
+    /**
+     * @return bool
+     */
+    public function isDistributedTracingActivationContext()
```
[the remark this answers is not in the document]
Comment (labbati) [reply src/DDTrace/SpanContext.php]: Added tot he property description of this is ok with you

File: src/DDTrace/Transport/Http.php (hunk 1 above)
Comment (labbati) [reply src/DDTrace/Transport/Http.php]: This is a good point and I like the idea. If you agree I will do for the parts that I touched in this PR and will postpone a wider refactoring of the Http transport (as the same applies also to other config params) to a later stage.

File: src/DDTrace/Transport/Http.php [hunk 3]
```diff
@@ -128,4 +137,30 @@ private function sendRequest($url, array $headers, $body)
...
+
+    /**
+     * Returns whether or not we should send these traces to the priority sampling aware trace agent endpoint.
+     * This approach could be optimized in the future if we refactor how traces are organized in parent/child relations
+     * but this would be out of scope at the moment.
+     *
+     * @param array $traces
+     * @return bool
+     */
+    private function isPrioritySamplingEndpoint(array $traces)
```
[the remark this answers is not in the document]
Comment (labbati) [reply src/DDTrace/Transport/Http.php]: totally 👍

File: src/DDTrace/Transport/Http.php (hunk 1 above)
Comment (labbati) [reply src/DDTrace/Transport/Http.php]: As discussed, it is risky as it introduces potential breaking changes for users, so I would defer it to a later http transport refactoring

**Comment by labbati** [comment]
@pawelchcki I did some manual testing and look good. It would be great to have your latest CR round. Thanks a lot!

**Pull request merged**